tools/disasm/sym_info.py

```diff
@@ -0,0 +1,217 @@
+#!/usr/bin/env python3
+# SPDX-FileCopyrightText: 2024 Dragorn421
+# SPDX-License-Identifier: CC0-1.0
+
+import argparse
+import csv
+import dataclasses
+from typing import Optional
+
+
+@dataclasses.dataclass
+class Sym:
+    name: str
+    value: int
+    type: Optional[str]
+    size: Optional[int]
+    vrom: int
+
+
+@dataclasses.dataclass
+class FileSection:
+    file: str
+    section: str
+    syms: list[Sym]
+    vma_start: int
+
+
+LABELS_TYPES = {
+    "@branchlabel",
+}
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Print informations on a symbol/address"
+        " (and possibly surrounding symbols)"
+        " from the spimdisasm disassembly context saved in"
+        " expected/build/VERSION/context.csv"
+    )
+    parser.add_argument("sym_or_vma")
+    default_version = "ntsc-1.2"
+    parser.add_argument(
+        "--version",
+        "-v",
+        default=default_version,
+        help=f"oot version (default: {default_version})",
+    )
+    parser.add_argument(
+        "--around",
+        "-n",
+        type=int,
+        default=0,
+        help="how many symbols to show around the target (at least)",
+    )
+    parser.add_argument(
+        "--range",
+        "-r",
+        type=lambda v: int(v, 0),
+        default=0,
+        help="show symbols within this range from the target (at least)",
+    )
+    parser.add_argument(
+        "--file",
+        "-f",
+        action="store_true",
+        help="show symbols within the file section of the target (at least)",
```

Comment:
I'm not sure what a "file section" is, maybe just spell out "symbols within the same file and same section as the target" or something?